### DISCUSSION TITLE
Dynamic QoS policy at memory controller according to PE id.

### DISCUSSION THREAD

**Discussion by Superstite** [gem5/gem5#1546] [Q&A]
Hi,
I have a network-on-chip setup with private L1 caches and shared L2 caches. I want to execute a dynamic QoS policy for the packets arriving at memory controllers. I know that there is a built-in QoS scheduler in gem5 which does something smilar. However when i extract the requestorID at schedule() in qos/mem_ctrl.cc, I got the ID of directory controllers and not the specific CPU/PE that generated the request. Can you advise me on how to get cpuid in the function schedule() in qos/mem_ctrl.cc? Further, how to make this policy dynamic, instead of static afetr analyzing data flow for a while.

**Comment by AgamemnonasKyriazis** [comment]
Hello - Sorry for posting on this after so much time but did you perhaps figure out how someone can use the requestorId to get the actual requestor, for example CPU etc. ?
